cmd/buildah/rmi.go

```diff
@@ -35,75 +47,192 @@ func rmiCmd(c *cli.Context) error {
+				if len(image.Names) > 1 && !force {
+					return fmt.Errorf("unable to delete %s (must force) - image is referred to in multiple tags", image.ID)
+				}
+				// If it is forced, we have to untag the image so that it can be deleted
+				image.Names = image.Names[:0]
+			} else {
+				name, err2 := untagImage(id, image, store)
+				if err2 != nil {
+					return err
 				}
+				fmt.Printf("untagged: %s", name)
+			}
+
+			if len(image.Names) > 0 {
+				continue
 			}
-			if ref != nil {
-				err = ref.DeleteImage(nil)
+			id, err := removeImage(image, store)
+			if err != nil {
+				return err
 			}
+			fmt.Printf("%s\n", id)
+		}
+	}
+
+	return nil
+}
+
+func getImage(id string, store storage.Store) (*storage.Image, error) {
+	var ref types.ImageReference
+	ref, err := properImageRef(id)
+	if err != nil {
+		//logrus.Debug(err)
+	}
+	if ref == nil {
+		if ref, err = storageImageRef(store, id); err != nil {
+			//logrus.Debug(err)
 		}
-		if e == nil {
-			e = err
+	}
+	if ref == nil {
+		if ref, err = storageImageID(store, id); err != nil {
+			//logrus.Debug(err)
 		}
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "error removing image %q: %v\n", id, err)
+	}
+	if ref != nil {
+		image, err2 := is.Transport.GetStoreImage(store, ref)
+		if err2 != nil {
+			return nil, err2
+		}
+		return image, nil
+	}
+	return nil, err
+}
+
+func untagImage(imgArg string, image *storage.Image, store storage.Store) (string, error) {
+	// Remove name from image.Names and set the new name in the ImageStore
+	imgStore, err := store.ImageStore()
+	if err != nil {
+		return "", errors.Wrap(err, "could not untag image")
+	}
+	newNames := []string{}
+	removedName := ""
+	for _, name := range image.Names {
+		if matchesReference(name, imgArg) {
+			removedName = name
 			continue
 		}
-		fmt.Printf("%s\n", id)
+		newNames = append(newNames, name)
 	}
+	imgStore.SetNames(image.ID, newNames)
+	err = imgStore.Save()
+	return removedName, err
+}
 
-	return e
+func removeImage(image *storage.Image, store storage.Store) (string, error) {
+	imgStore, err := store.ImageStore()
+	if err != nil {
+		return "", errors.Wrapf(err, "could not open image store")
+	}
+	err = imgStore.Delete(image.ID)
+	if err != nil {
+		return "", errors.Wrapf(err, "could not remove image")
+	}
+	err = imgStore.Save()
+	if err != nil {
+		return "", errors.Wrapf(err, "could not save image store")
+	}
+	return image.ID, nil
+}
+
+// Returns a list of running containers associated with the given ImageReference
+func runningContainers(image *storage.Image, store storage.Store) ([]string, error) {
+	ctrIDs := []string{}
+	ctrStore, err := store.ContainerStore()
+	if err != nil {
+		return nil, err
+	}
+
+	containers, err := ctrStore.Containers()
+	if err != nil {
+		return nil, err
+	}
+	for _, ctr := range containers {
+		if ctr.ImageID == image.ID {
+			ctrIDs = append(ctrIDs, ctr.ID)
+		}
+	}
+	return ctrIDs, nil
+}
+
+func removeRunningContainers(ctrIDs []string, store storage.Store) error {
```

Comment:
Should this be named 'removeContianers'?  I'm not seeing any checks to see if the container is running within the function.